tests/RobotFramework/tests/mqtt/mosquitto_bug_check.robot

```diff
@@ -0,0 +1,38 @@
+*** Settings ***
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+Library             String
+
+Suite Setup         Setup
+Suite Teardown      Get Logs
+
+
+*** Test Cases ***
+Stop tedge-mapper-c8y
+    Execute Command    sudo systemctl stop tedge-mapper-c8y.service
+
+Publish an MQTT message with QoS 1
+    Execute Command    tedge mqtt pub -q 1 te/device/main///e/test "{\\"text\\":\\"hello $(date +%s)\\"}"
+
+Start tedge-mapper-c8y
+    Execute Command    sudo systemctl start tedge-mapper-c8y.service
+
+Check result
+    ${result} =    Execute Command    mosquitto_sub -h 127.0.0.1 -c -i subscriber1 -t topic1 --qos 1 -W 2; mosquitto_pub -h 127.0.0.1 -t topic1 -m test --qos 1; sleep 1; mosquitto_sub -h 127.0.0.1 -c -i subscriber1 -t topic1 --qos 1 -C 1 -W 2 && echo PASSED || echo FAILED;
+    ${trimmed_result} =    Remove Extra Spaces And Newlines    ${result}
+    Log Result    ${trimmed_result}
```

Comment:
Just create 1 test case rather the splitting the test setup into individual test cases, and don't bother testing for the local mqtt persistence, just check if the message is delivered to the cloud or not (as this is the user contract).

So the (single) test case should do:

1. Stop the tedge-mapper-c8y
2. Publish an event
3. Start the tedge-mapper-c8y
4. Check that the event is visible in the cloud